blueman/main/Adapter.py

```diff
@@ -84,8 +74,21 @@ def __init__(self, selected_hci_dev, socket_id):
                 logging.error('Error: the selected adapter does not exist')
         self.notebook.show_all()
 
-    def _on_close(self, *args, **kwargs):
-        Gtk.main_quit()
+    def do_activate(self):
+        def app_release(_plug, event):
+            self.release()
+
+        if self.socket_id:
+            self.hold()
+            plug = Gtk.Plug.new(self.socket_id)
+            plug.show()
+            plug.connect('delete-event', app_release)
+            plug.add(self.notebook)
+        else:
+            if not self.window:
+                self.window = BluemanAdaptersWindow(app=self)
+                self.window.add(self.notebook)
+            self.window.present_with_time(Gtk.get_current_event_time())
```

Comment:
This is only necessary if `self.window` did exist already, no? The different `Gtk.Application`s differ in usages. `Blueman.do_activate` and `BluemanServices.do_activate` do not call it for new windows.

Reply:
I blame the reverse logic for this :fearful:. But there should not be any difference here between all the apps, I'll change it so it works like the others.